docs: getting started zkApps (#632)
o1-labs/docs2: docs/zkapps/getting-started-zkapps.mdx

See [zkApp CLI Installation](/zkapps/install-zkapp-cli).

### 2. Create a project

```sh
zk project <project-name>
```

[o1js](/zkapps/o1js) is automatically installed when you generate a project using the zkApp CLI. 

A zkApp consists of a smart contract and a UI to interact with it. 

- To proceed without an accompanying UI project, select `none` when prompted.

    See [Option B: Start your own project](/zkapps/how-to-write-a-zkapp#option-b-start-your-own-project).

- To create a UI, select a framework and follow the prompts. 

    See [How to Write a zkApp UI](/zkapps/how-to-write-a-zkapp-ui).

### 3. Add testing code

When you use the zkApp CLI to create a project, tests and examples are included. 

1. See the `import` statements in the [Add.test.ts](https://github.com/o1-labs/zkapp-cli/blob/main/templates/project-ts/src/Add.test.ts#L1-L2) example file.

1. A `LocalBlockchain` instance to interact with is included in the [Add.test.ts](https://github.com/o1-labs/zkapp-cli/blob/main/templates/project-ts/src/Add.test.ts#L27-L28) example file.

> 1. In o1js, an array of 10 test accounts to pay fees and sign transactions are provided for the `LocalBlockchain` instance in the `Local.testAccounts;` in the [Add.test.ts](https://github.com/o1-labs/zkapp-cli/blob/main/templates/project-ts/src/Add.test.ts#L29-L31) example file. The example accesses two of these public/private key pair accounts. The example uses these names, but the names can be anything:

Reviewer: ```suggestion
1. In o1js, an array of 10 test accounts to pay fees and sign transactions are provided for the `LocalBlockchain` instance. These can be accessed with `Local.testAccounts` as per the [Add.test.ts](https://github.com/o1-labs/zkapp-cli/blob/main/templates/project-ts/src/Add.test.ts#L29-L31) example file. The example uses the public/private key pairs of two of these accounts. They are given these names, but the names can be anything:
```

    - `deployerAccount` deploys the smart contract
    - `senderAccount` pays transaction fees

1. Deploy the smart contract to the `LocalBlockchain` instance that simulates a network for testing purposes.

    See the `localDeploy` function in the [Add.test.ts](https://github.com/o1-labs/zkapp-cli/blob/main/templates/project-ts/src/Add.test.ts#L38-L46) example file.

### 4. Add functionality

Add the logic for your smart contract.

> 1. Start experimenting with iterative development to build and test one method at a time. Add functionality to the smart contract by implementing an `@method`.

Reviewer: ```suggestion
1. Start experimenting with iterative development to build and test one method at a time. Add functionality to the smart contract by implementing a `@method`.
```
I think? Feels better to me anyway

    See `@method update()` in the [Add.ts](https://github.com/o1-labs/zkapp-cli/blob/main/templates/project-ts/src/Add.ts#L20-L24) example file.

1. Build the smart contract:

    ```sh
    npm run build
    ```

1. Invoke the `@method` you added or use new functionality in the test file.

    See the transaction code that invokes the `update()` method in the [Add.test.ts](https://github.com/o1-labs/zkapp-cli/blob/main/templates/project-ts/src/Add.test.ts#L57-L66) file.

    - If it works as expected with no errors, add more functionality.
    - If there are errors, look through the stack traces to find the source of the errors and update the contract to resolve them.

### 5. Create an integration test.

- Create a Node.js script to run the smart contract and test it's functionality, similar to step [3. Add testing code](#3-add-testing-code).

    For an example, see the Node.js script that runs the Tic Tac Toe smart contract in the [run.ts](https://github.com/o1-labs/zkapp-cli/blob/main/examples/tictactoe/ts/src/run.ts) file.

### 6. Test with a live network

To deploy the smart contract to the Testnet, run the `zk` commands from the directory that contains your smart contract. 

> 1. Configure your zkapp.

Reviewer: ```suggestion
1. Configure your zkApp.
```

- To create a UI, select a framework and follow the prompts. 

See [How to Write a zkApp UI](/zkapps/how-to-write-a-zkapp-ui).

### 3. Add smart contract code

Add the smart contract code to your zkApp project using the [o1js](/zkapps/o1js) TypeScript library that was included automatically when you created your project using the zkApp CLI.

Start by reviewing or using the basic smart contract example that was generated when you created your project.

1. Import `o1js`. 

    See `import` in the [Add.ts](https://github.com/o1-labs/zkapp-cli/blob/main/templates/project-ts/src/Add.ts) file.

1. Extend the `SmartContract` class.

    See `export` in the [Add.ts](https://github.com/o1-labs/zkapp-cli/blob/main/templates/project-ts/src/Add.ts#L12) file. 

See [Smart Contracts](https://docs.minaprotocol.com/zkapps/o1js/smart-contracts).

### 4. Add testing code

When you use the zkApp CLI to create a project, tests and examples are included.

1. Import the smart contract and `o1js` into your test file. 

    See the `import` statements in the [Add.test.ts](https://github.com/o1-labs/zkapp-cli/blob/main/templates/project-ts/src/Add.test.ts#L1-L2) example file.

> 1. Create a local blockchain instance to interact with.

Author: @nicc  happy to make changes when I know something is a code literal - sometimes, I just don't know (yet)

    See the local blockchain code in the [Add.test.ts](https://github.com/o1-labs/zkapp-cli/blob/main/templates/project-ts/src/Add.test.ts#L27-L28) example file.

1. Initialize test accounts to pay fees and sign transactions. o1js provides an array of 10 accounts you can start using right away.

    See the initialize account code in the [Add.test.ts](https://github.com/o1-labs/zkapp-cli/blob/main/templates/projec%C3%A5%C3%A7t-ts/src/Add.test.ts#L29-L31) example file.

1. Deploy the smart contract to the local blockchain.

>     See [Deploying a contract locally](https://docs.minaprotocol.com/zkapps/how-to-test-a-zkapp#deploying-a-contract-locally).

Author: TIL copy-pasta
